### PR TITLE
docs(signals): add FAQ

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

I skimmed through discord/twitter/discussions to search frequently asked questions.
This page can start as a starting point, which we can extend later.

I didn't include the question "what's the difference with store / component store", because I think it would be better to give that it's own page. We could move https://ngrx.io/guide/component-store/comparison from a subpage to an index page and include signals. This way it gets more visibility and it includes all NgRx packages.

Preview: https://deploy-preview-4193--ngrx-io.netlify.app/guide/signals/faq

### CODE REVIEW

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<details>
+  <summary>How to connect my signal store(s) with Redux DevTools?</summary>
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:2]:
```suggestion
  <summary>How to connect my SignalStore(s) with Redux DevTools?</summary>
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    There's no official connection between `@ngrx/signals` and the Redux Devtools.
+    We expect the Angular Devtools will provide support for signals soon, which can be used to track the state.
+    However, you could create a feature for this, or you can make use of the [`withDevtools` feature](https://github.com/angular-architects/ngrx-toolkit?tab=readme-ov-file#devtools-withdevtools) from angular-architects/ngrx-toolkit.
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:6]:
```suggestion
    However, you could create a feature for this, or you can make use of the [`withDevtools` feature](https://github.com/angular-architects/ngrx-toolkit?tab=readme-ov-file#devtools-withdevtools) from the `@angular-architects/ngrx-toolkit` package.
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+</details>
+
+<details>
+  <summary>Is there a way to define private methods on a signal store?</summary>
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:10]: for consistency:

```suggestion
  <summary>Is there a way to define private methods on a SignalStore?</summary>
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    publicFunction() { 
+        /* implementation here */
+    }
+    publicFunction2() { 
+        // it's possible to invoke private methods
+        privateFunction();
+    }
+
+    return { publicFunction, publicFunction2 };
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:28]:
```suggestion
withMethods(() => {
  function privateFunction() {
    /* implementation here */ 
  }
  function publicFunction() {
    /* implementation here */
  }
  function publicFunction2() {
    privateFunction();
  }

  return { publicFunction, publicFunction2 };
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+})
+```
+</details>
+
+<details>
+  <summary>Can I  interact with my NgRx Actions within a signal store?</summary>
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:34]:
```suggestion
  <summary>Can I  interact with my NgRx Actions within a SignalStore?</summary>
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    Signals are not meant to have a concept of time. Also, the effect is somewhat tied to Angular change detection, so you can't observe every action that would be dispatched over time through some sort of Signal API.
+    The global NgRx store is still the best mechanism to dispatch action(s) over time and react to them across multiple features.
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:37]:
```suggestion
    The global NgRx Store is still the best mechanism to dispatch action(s) over time and react to them across multiple features.
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+</details>
+
+<details>
+  <summary>Can I use the Redux pattern (reducers) to build my state?</summary>
+
+    Just like @ngrx/component-store, there is no indirection between events and how it affects the state. To update the signal store's state use the methods provided by [`SignalStore`](/guide/signals/signal-store) and [`SignalState`](/guide/signals/signal-state).
+    But, a signal store is extensible and you can build your own custom feature that uses the Redux pattern.
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:44]:
```suggestion
    Just like `@ngrx/component-store`, there is no indirection between events and how it affects the state. To update the SignalStore's state use the `patchState` function.
    However, SignalStore is extensible and you can build your own custom feature that uses the Redux pattern. 
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  <summary>Can I use the Redux pattern (reducers) to build my state?</summary>
+
+    Just like @ngrx/component-store, there is no indirection between events and how it affects the state. To update the signal store's state use the methods provided by [`SignalStore`](/guide/signals/signal-store) and [`SignalState`](/guide/signals/signal-state).
+    But, a signal store is extensible and you can build your own custom feature that uses the Redux pattern. 
+</details>
+
+<details>
+  <summary>Can I define my signal store as a class?</summary>
+
+    To create a class-based signal store, create a new class and extend from `signalStore`.
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:50]:
```suggestion
<details>
  <summary>Can I define my SignalStore as a class?</summary>

    To create a class-based SignalStore, create a new class and extend from `signalStore`.
```

File: projects/ngrx.io/content/guide/signals/faq.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+```ts
+export MyStore extends signalStore(withState(initialState)) {
+    private myMethod() {}
+}
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/signals/faq.md:55]:
```suggestion
@Injectable()
export class CounterStore extends signalStore(withState({ count: 0 })) {
  readonly doubleCount = computed(() => this.count() * 2);
  
  increment(): void {
    patchState(this, { count: this.count() + 1 });
  }
}
```

**Review by timdeschryver — COMMENTED** [review]
Thanks for the review!

**Review by markostanimirovic — APPROVED** [review]
🚀

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**